lib/cookieContentHandler.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ *  All Rights Reserved
+ *  MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+ * OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT
+ * OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+'use restrict';
+
+var crypto = require('crypto');
+var createBuffer = require('./jwe').createBuffer;
+
+/*
+ * the handler for state/nonce/policy
+ * @maxAmount          - the max amount of {state: x, nonce: x, policy: x} tuples you want to save in cookie
+ * @maxAge            - when a tuple in session expires in seconds
+ * @cookieEncryptionKeys  
+ *                    - keys used to encrypt and decrypt cookie
+ */ 
+function CookieContentHandler(maxAmount, maxAge, cookieEncryptionKeys) {
+  if (!maxAge || (typeof maxAge !== 'number' || maxAge <= 0))
+    throw new Error('CookieContentHandler: maxAge must be a positive number');
+  this.maxAge = maxAge;  // seconds
+
+  if (!maxAmount || (typeof maxAmount !== 'number' || maxAmount <= 0 || maxAmount % 1 !== 0))
+    throw new Error('CookieContentHandler: maxAmount must be a positive integer');
+  this.maxAmount = maxAmount;
+
+  if (!cookieEncryptionKeys || !Array.isArray(cookieEncryptionKeys) || cookieEncryptionKeys.length === 0)
+    throw new Error('CookieContentHandler: cookieEncryptionKeys must be a non-emptry array');
+
+  for (var i = 0; i < cookieEncryptionKeys.length; i++) {
+    var item = cookieEncryptionKeys[i];
+    if (!item.key || !item.iv)
+      throw new Error(`CookieContentHandler: array item ${i+1} in cookieEncryptionKeys must have the form { key: , iv: }`);
+    if (item.key.length !== 32)
+      throw new Error(`CookieContentHandler: key number ${i+1} is ${item.key.length} bytes, expected: 32 bytes`);
```

Comment:
@lovemaths 

Why would the key need to be exactly 32 bytes ?

The [cookie-session](https://www.npmjs.com/package/cookie-session) seems to accept secret/keys of any size.